packages/artifact/src/upload-http-client.ts

```diff
@@ -0,0 +1,326 @@
+  } else {
+    // eslint-disable-next-line no-console
+    console.log(rawResponse)
+    throw new Error(
+      `Unable to create a container for the artifact ${artifactName}`
+    )
+  }
+}
+
+/**
+ * Concurrently upload all of the files in chunks
+ * @param {string} uploadUrl Base Url for the artifact that was created
+ * @param {SearchResult[]} filesToUpload A list of information about the files being uploaded
+ * @returns The size of all the files uploaded in bytes
+ */
+export async function uploadArtifactToFileContainer(
+  uploadUrl: string,
+  filesToUpload: UploadSpecification[],
+  options?: UploadOptions
+): Promise<UploadResults> {
+  const client = createHttpClient()
+  const FILE_CONCURRENCY = getUploadFileConcurrency()
+  const CHUNK_CONCURRENCY = getUploadChunkConcurrency()
+  const MAX_CHUNK_SIZE = getUploadChunkSize()
+  debug(
+    `File Concurrency: ${FILE_CONCURRENCY}, Chunk Concurrency: ${CHUNK_CONCURRENCY} and Chunk Size: ${MAX_CHUNK_SIZE}`
+  )
+
+  const parameters: UploadFileParameters[] = []
+
+  // by default, file uploads will continue if there is an error unless specified differently in the options
+  let continueOnError = true
+  if (options) {
+    if (options.continueOnError === false) {
+      continueOnError = false
+    }
+  }
+
+  // Prepare the necessary parameters to upload all the files
+  for (const file of filesToUpload) {
+    const resourceUrl = new URL(uploadUrl)
+    resourceUrl.searchParams.append('itemPath', file.uploadFilePath)
+    parameters.push({
+      file: file.absoluteFilePath,
+      resourceUrl: resourceUrl.toString(),
+      restClient: client,
+      concurrency: CHUNK_CONCURRENCY,
+      maxChunkSize: MAX_CHUNK_SIZE,
+      continueOnError
+    })
+  }
+
+  const parallelUploads = [...new Array(FILE_CONCURRENCY).keys()]
+  const failedItemsToReport: string[] = []
+  let uploadedFiles = 0
+  let fileSizes = 0
+  let abortPendingFileUploads = false
+
+  // Only allow a certain amount of files to be uploaded at once, this is done to reduce potential errors
+  await Promise.all(
+    parallelUploads.map(async () => {
+      while (uploadedFiles < filesToUpload.length) {
+        const currentFileParameters = parameters[uploadedFiles]
+        uploadedFiles += 1
+        if (abortPendingFileUploads) {
+          failedItemsToReport.push(currentFileParameters.file)
+          continue
+        }
+
+        const uploadFileResult = await uploadFileAsync(currentFileParameters)
+        fileSizes += uploadFileResult.successfulUploadSize
+        if (uploadFileResult.isSuccess === false) {
+          failedItemsToReport.push(currentFileParameters.file)
+          if (!continueOnError) {
+            // Existing uploads will be able to finish however all pending uploads will fail fast
+            abortPendingFileUploads = true
+          }
+        }
+      }
+    })
+  )
+
+  // eslint-disable-next-line no-console
+  console.log(`Total size of all the files uploaded is ${fileSizes} bytes`)
```

Comment:
Minor: you can use `core.info` here and elsewhere if you want to `console.log`. It's easier to mock `core` than `console` and you don't need the eslint disables

Reply:
It seems like `core.info` isn't documented anywhere, even though it exists 🤔 https://github.com/actions/toolkit/tree/master/packages/core

I can use it a few places, though `console.log` is better because it isn't limited to just strings so certain things like formatted http responses look better